src/ASM/LR/ASMu2Drecovery.C

```diff
@@ -426,7 +428,8 @@ LR::LRSplineSurface* ASMu2D::regularInterpolation (const RealArray& upar,
     // optimization note:
     // without an element id, splineValues will be stored as a full dense vector
     for (size_t j = 0; j < nBasis; j++)
-      A(i+1,j+1) = splineValues.basisValues[j];
+      if (utl::trunc(splineValues.basisValues[j]) != 0.0)
```

Comment:
-- original --
ASMu2D::projectSolutio|    542.10 77443.39 |    185.94 26562.60 |     7

Peak memory usage: 3.2GB

-- new, mine --
ASMu2D::integrate(I)  |     38.90  1852.45 |     37.72  1796.08 |    2

Peak memory usage: 220MB

-- new, kjetil --
ASMu2D::projectSolutio|      2.49   355.38 |      2.49   355.52 |     7

Peak memory usage: 220MB

-- new, kjetil, old liblr --
ASMu2D::projectSolutio|    152.47 21780.98 |    152.61 21802.05 |     7
Peak memory usage: 220MB

last result just to  show why i didn't do it like this prior to the optimization.